Autopilot/boardfiles/Src/freertos.c

```diff
@@ -136,40 +160,143 @@ void MX_FREERTOS_Init(void) {
+  InterchipHandle = osThreadCreate(osThread(interchip), NULL);
+
+  /* definition and creation of pathManager */
+  osThreadDef(pathManager, pathManagerExecute, osPriorityNormal, 0, 128);
+  pathManagerHandle = osThreadCreate(osThread(pathManager), NULL);
+
+  /* definition and creation of telemetryRun */
+  osThreadDef(telemetryRun, StartTelemetryRun, osPriorityNormal, 0, 128);
+  telemetryRunHandle = osThreadCreate(osThread(telemetryRun), NULL);
 
-  /* definition and creation of Attitude */
-  osThreadDef(Attitude, Attitude_Run, osPriorityNormal, 0, 128);
-  AttitudeHandle = osThreadCreate(osThread(Attitude), NULL);
+   /* definition and creation of sensorFusionRun */
+  osThreadDef(sensorFusionRun, SensorFusionExecute, osPriorityNormal, 0, 128);
+  sensorFusionHandle = osThreadCreate(osThread(sensorFusionRun), NULL);
+
+  osThreadDef(testThread, testExecute, osPriorityNormal, 0, 128);
+  test = osThreadCreate(osThread(testThread), NULL);
+
+
+  /* definition and creation of sensorFusionRun */
 
   /* USER CODE BEGIN RTOS_THREADS */
   /* add threads, ... */
   /* USER CODE END RTOS_THREADS */
 
 }
 
-/* USER CODE BEGIN Header_StartDefaultTask */
+/* USER CODE BEGIN Header_attitudeManagerExecute */
 /**
-  * @brief  Function implementing the defaultTask thread.
+  * @brief  Function implementing the attitudeManager thread.
   * @param  argument: Not used
   * @retval None
   */
-/* USER CODE END Header_StartDefaultTask */
-void StartDefaultTask(void const * argument)
+/* USER CODE END Header_attitudeManagerExecute */
+void attitudeManagerExecute(void const * argument)
+{
+  /* USER CODE BEGIN attitudeManagerExecute */
+  /* Infinite loop */
+  while(1)
+  {
+    TickType_t xLastWakeTime = xTaskGetTickCount();
+    vTaskDelayUntil(&xLastWakeTime, PERIOD_ATTITUDEMANAGER_MS);
+    bool status = AttitudeManagerInterfaceExecute();
+    if (!status) {
+      catastrophicFailure = true;
+    }
+    HAL_GPIO_TogglePin(LED1_GPIO_Port, LED1_Pin);
+  }
+  /* USER CODE END attitudeManagerExecute */
+}
+
+/* USER CODE BEGIN Header_pathManagerExecute */
+/**
+* @brief Function implementing the pathManager thread.
+* @param argument: Not used
+* @retval None
+*/
+/* USER CODE END Header_pathManagerExecute */
+void pathManagerExecute(void const * argument)
+{
+  /* USER CODE BEGIN pathManagerExecute */
+  /* Infinite loop */
+  while(1)
+  {
+    TickType_t xLastWakeTime = xTaskGetTickCount();
+    vTaskDelayUntil(&xLastWakeTime, PERIOD_PATHMANAGER_MS);
+    bool status = PathManagerInterfaceExecute();
+    if (!status) {
+      catastrophicFailure = true;
+    }
+    HAL_GPIO_TogglePin(LED2_GPIO_Port, LED2_Pin);
+  }
+  
+  /* USER CODE END pathManagerExecute */
+}
+
+/* USER CODE BEGIN Header_StartTelemetryRun */
+/**
+* @brief Function implementing the telemetryRun thread.
+* @param argument: Not used
+* @retval None
+*/
+/* USER CODE END Header_StartTelemetryRun */
+void StartTelemetryRun(void const * argument)
```

Comment:
any chance we can name this telemetryManagerExecute for consistency

Reply:
Will do